File: CS/WindowsFormsApplication202/MyImageSlider.cs
using DevExpress.Utils;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.CustomEditor;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;

namespace WindowsFormsApplication202 {
    class MyImageSlider : ImageSlider, IAnyControlEdit {

        public MyImageSlider( ) {
            this.Size = new Size(200, 100);
            this.LayoutMode = DevExpress.Utils.Drawing.ImageLayoutMode.MiddleCenter;
        }
        protected override void OnLeave(EventArgs e) {
            base.OnLeave(e);
                UpdateEditValue();
        }
        private void UpdateEditValue() {
            if (EditValueChanged != null)
                EditValueChanged(this, EventArgs.Empty);
        }
        
        public bool AllowBitmapCache {
            get { return true; }
        }

        public bool AllowBorder {
            get { return false; }
        }

        public bool AllowClick(System.Drawing.Point point) {
            return true;
        }

        public System.Drawing.Size CalcSize(System.Drawing.Graphics g) {
            return this.Size; 
        }

        public void Draw(DevExpress.Utils.Drawing.GraphicsCache cache, AnyControlEditViewInfo viewInfo) {
            throw new NotImplementedException();
        }

        Image fEditValue = null;
        public object EditValue {
            get {
                return fEditValue;
            }
            set {
                Image itemValue = value as Image;
                if (itemValue != null) {
                    fEditValue = itemValue;
                    this.Images.Clear();
                    this.Images.Add(fEditValue);
                    this.CurrentImageIndex = 0;
                    if (ImageList != null) {
                    }
                }
            }
        }

        public event EventHandler EditValueChanged;

        public string GetDisplayText(object EditValue) {
            return EditValue == null ? string.Empty : EditValue.ToString();
        }

        public bool IsNeededKey(System.Windows.Forms.KeyEventArgs e) {
            return false;
        }

        public void SetupAsDrawControl() {
            
        }

        public void SetupAsEditControl() {
            throw new NotImplementedException();
        }

        public bool SupportsDraw {
            get { return false; }
        }

        private void InitializeComponent() {
            ((System.ComponentModel.ISupportInitialize)(this)).BeginInit();
            this.SuspendLayout();
            // 
            // MyImageSlider
            // 
            this.CurrentImageIndex = -1;
            ((System.ComponentModel.ISupportInitialize)(this)).EndInit();
            this.ResumeLayout(false);

        }
    }
}

File: VB/WindowsFormsApplication202/MyImageSlider.vb
Imports DevExpress.Utils
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.CustomEditor
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text

Namespace WindowsFormsApplication202
	Friend Class MyImageSlider
		Inherits ImageSlider
		Implements IAnyControlEdit

		Public Sub New()
			Me.Size = New Size(200, 100)
			Me.LayoutMode = DevExpress.Utils.Drawing.ImageLayoutMode.MiddleCenter
		End Sub
		Protected Overrides Sub OnLeave(ByVal e As EventArgs)
			MyBase.OnLeave(e)
				UpdateEditValue()
		End Sub
		Private Sub UpdateEditValue()
			RaiseEvent EditValueChanged(Me, EventArgs.Empty)
		End Sub

		Public ReadOnly Property AllowBitmapCache() As Boolean Implements IAnyControlEdit.AllowBitmapCache
			Get
				Return True
			End Get
		End Property

		Public ReadOnly Property AllowBorder() As Boolean Implements IAnyControlEdit.AllowBorder
			Get
				Return False
			End Get
		End Property

		Public Function AllowClick(ByVal point As System.Drawing.Point) As Boolean Implements IAnyControlEdit.AllowClick
			Return True
		End Function

		Public Function CalcSize(ByVal g As System.Drawing.Graphics) As System.Drawing.Size Implements IAnyControlEdit.CalcSize
			Return Me.Size
		End Function

		Public Sub Draw(ByVal cache As DevExpress.Utils.Drawing.GraphicsCache, ByVal viewInfo As AnyControlEditViewInfo) Implements IAnyControlEdit.Draw
			Throw New NotImplementedException()
		End Sub

		Private fEditValue As Image = Nothing
		Public Property EditValue() As Object Implements IAnyControlEdit.EditValue
			Get
				Return fEditValue
			End Get
			Set(ByVal value As Object)
				Dim itemValue As Image = TryCast(value, Image)
				If itemValue IsNot Nothing Then
					fEditValue = itemValue
					Me.Images.Clear()
					Me.Images.Add(fEditValue)
					Me.CurrentImageIndex = 0
					If ImageList IsNot Nothing Then
					End If
				End If
			End Set
		End Property

		Public Event EditValueChanged As EventHandler Implements IAnyControlEdit.EditValueChanged

		Public Function GetDisplayText(ByVal EditValue As Object) As String Implements IAnyControlEdit.GetDisplayText
			Return If(EditValue Is Nothing, String.Empty, EditValue.ToString())
		End Function

		Public Function IsNeededKey(ByVal e As System.Windows.Forms.KeyEventArgs) As Boolean Implements IAnyControlEdit.IsNeededKey
			Return False
		End Function

		Public Sub SetupAsDrawControl() Implements IAnyControlEdit.SetupAsDrawControl

		End Sub

		Public Sub SetupAsEditControl() Implements IAnyControlEdit.SetupAsEditControl
			Throw New NotImplementedException()
		End Sub

		Public ReadOnly Property SupportsDraw() As Boolean Implements IAnyControlEdit.SupportsDraw
			Get
				Return False
			End Get
		End Property

		Private Sub InitializeComponent()
			DirectCast(Me, System.ComponentModel.ISupportInitialize).BeginInit()
			Me.SuspendLayout()
			' 
			' MyImageSlider
			' 
			Me.CurrentImageIndex = -1
			DirectCast(Me, System.ComponentModel.ISupportInitialize).EndInit()
			Me.ResumeLayout(False)

		End Sub
	End Class
End Namespace
